lib/archethic/beacon_chain.ex

```diff
@@ -84,6 +84,32 @@ defmodule ArchEthic.BeaconChain do
     end)
   end
 
+  @doc """
+  Retrieve the beacon summaries storage nodes from two dates
+  """
+  @spec get_summary_pool(DateTime.t(), DateTime.t()) :: pools()
+  def get_summary_pool(
```

Comment:
Why do you need to have two date to get the storage nodes for the beacon summary ?

Reply:
when I get that from the one date it is taking a lot of time,

    summary_times1 = SummaryTimer.previous_summaries(last_sync_date)
    summary_times2 = SummaryTimer.previous_summaries(date)
    summary_times = summary_times1 -- summary_times2

so, this what I did to overcome that.

Reply:
Once you have a date, you should not iterate over the dates anymore.
Just perform election based on the subsets and the date, and you should have all the nodes.
So no need to leverage `SummaryTimer.previous_summaries` anymore.